#1049: Adding new doctypes

alw-bsz opened the pull request:
Adding publish templates, translations, and ini-params for new doctypes

j3nsch committed:
#153 Modified Solr setup for new Solarium

j3nsch committed:
#153 Modified Vagrant setup for PHP 7.4

j3nsch committed:
#153 Reverse removal of setup of console.ini

j3nsch committed:
#153 Setup GitHub testing with PHP 7.1 and 7.4

j3nsch committed:
#153 Disable PHP source lint check

j3nsch committed:
#153 Activate lint again, trying to fix problem

j3nsch committed:
#153 Problem line commented out

j3nsch committed:
#153 Still analysing lint problem

j3nsch committed:
#153 Problem test removed (temporary)

j3nsch committed:
#153 Test added again

j3nsch committed:
#153 Testing problem withinline text

j3nsch committed:
#153 Trying another inline text solution

j3nsch committed:
#153 Fixing PHP linting problem

j3nsch committed:
#153 Fixed test expectations

j3nsch committed:
#153 Coding style fixes

j3nsch committed:
#153 Coding style fixes for PHP 7.1
Remove time limit for coding style check in composer.json

j3nsch committed:
#153 Coding style fixes

j3nsch committed:
#153 Coding style and test fix

j3nsch committed:
#153 Fix parameters passed by reference

j3nsch committed:
#153 Fix coding style

j3nsch committed:
#153 Automatic testing with PHP 7.2

j3nsch committed:
#995 Support for module CSS file

j3nsch committed:
#995 Only test with PHP 7.1 for now

j3nsch committed:
#995 Update branch from 4.7.2

j3nsch committed:
#995 Fix return type

j3nsch committed:
#996 Add 'required' CSS class to entire form-item

j3nsch committed:
Merge branch '4.7.2' into css996

j3nsch committed:
Merge branch '4.7.2' into css996

j3nsch committed:
#1034 Removed CSS for <IE9

j3nsch committed:
#995 Started config for CSS-themes

j3nsch committed:
#996 Merge v4.8.1 changes

j3nsch committed:
#995 Draft for new CSS configuration

j3nsch committed:
#996 Added 'viewport' meta tag

j3nsch committed:
#996 Added SPAN element around selected sorting option

j3nsch committed:
#996 Added SPAN element around 'sorting_relevancy'

j3nsch committed:
#996 fonts added

j3nsch committed:
#996 Do not ignore 'components'

j3nsch committed:
#996 Updated fontawesome

j3nsch committed:
#996 Added new CSS files

j3nsch committed:
#996 Formatting fixes

j3nsch committed:
#996 Use new CSS

j3nsch committed:
#996 Update opus4-common dependency

j3nsch committed:
#996 Changed DOCTYPE back because of validation errors

j3nsch committed:
#996 Fixed coding style

vgerlach committed:
Update frontdoor.css

vgerlach committed:
Update fontawesome--custom.css

j3nsch committed:
Merge pull request #1043 from vgerlach/patch-1

Update new css files

vgerlach committed:
Update fontawesome--custom.css

Beim Durchsehen einzelner Seiten-Ausgaben haben wir festgestellt, dass die Breadcrumb navigation im Search-Bereich noch nicht gestylt war. Diese Navigation erscheint nur in einem bestimmten Kontext.

vgerlach committed:
Update search.css

Beim Durchsehen einzelner Seiten-Ausgaben haben wir festgestellt, dass die Breadcrumb navigation im Search-Bereich noch nicht gestylt war. Diese Navigation erscheint nur in einem bestimmten Kontext.

j3nsch committed:
Merge pull request #1044 from vgerlach/vgerlach-css-update

Breadcrumbs on search pages fixed

vgerlach committed:
Update publish.css

Optische Trennung von mehreren AutorInnen, GutachterInnen, BetreuerInnen... beim Veröffentlichen

vgerlach committed:
Update theme.css

Optische Trennung von mehreren AutorInnen, GutachterInnen, BetreuerInnen... beim Veröffentlichen

vgerlach committed:
Merge pull request #1 from vgerlach/patch-3

Update theme.css

j3nsch committed:
Merge pull request #1058 from vgerlach/patch-2

Update publish.css and theme.css

vgerlach committed:
Update base.css

vgerlach committed:
Update search.css

vgerlach committed:
Update admin--overrides.css

vgerlach committed:
Update frontdoor.css

vgerlach committed:
Update fontawesome--custom.css

u.a. Abstract Ein- und Ausklappen

vgerlach committed:
Update search.css

vgerlach committed:
Update frontdoor.css

vgerlach committed:
Update publish.css

vgerlach committed:
Update theme.css

j3nsch committed:
Merge pull request #1069 from vgerlach/patch-7

Update admin--overrides.css

j3nsch committed:
Merge pull request #1070 from vgerlach/patch-5

Update search.css

j3nsch committed:
Merge pull request #1071 from vgerlach/patch-4

Update base.css

j3nsch committed:
Merge pull request #1072 from vgerlach/patch-8

Update frontdoor.css

j3nsch committed:
Merge pull request #1073 from vgerlach/patch-9

Update fontawesome--custom.css

j3nsch committed:
Merge pull request #1074 from vgerlach/patch-10

Update search.css

j3nsch committed:
Merge pull request #1075 from vgerlach/patch-11

Update frontdoor.css

j3nsch committed:
Merge pull request #1077 from vgerlach/patch-13

Update theme.css

j3nsch committed:
#1056 Removed 300px inline style

vgerlach committed:
Update publish.css

Es bestand der Wunsch, die Select-Boxen so breit zu gestalten, dass die Ausgangs-Option visuell nicht abgeschnitten wird, wenn sie mehr Platz benötigt als 300px. Daher hatten wir kürzlich die 300px-Angabe in unserem CSS mit automatischer Breite überschrieben.

Allerdings gibt es auswählbare Optionen in den Selects, die gefühlt ins Nebenzimmer laufen. ;-) Und die Selects passen sich ohne Breitenangabe der längsten Option an. Prozentuale Angaben funktionieren nicht, weil sie sich auf die längste Option beziehen. Damit erschließt sich auch, weshalb hier von Ihrer Entwicklung eine Breitenangabe hard codiert als Inline Style angegeben wird.

Ich habe diese Breite jetzt ins CSS eingetragen. Der Inline-Style sollte damit nicht mehr nötig sein und kann aus dem generierenden Code entfernt werden. Generell sollte wegen der CSS Kaskadierungsregeln auf Inline-Styles verzichtet werden, sie lassen sich nur noch mit !important im CSS überschreiben und bestimmen damit das Ende der Überschreibungsmöglichkeiten. Also möglichst komplette Trennung von HTML-Markup und CSS-Layout. ;-)

j3nsch committed:
Merge pull request #1076 from vgerlach/patch-12

Update publish.css

j3nsch committed:
#996 Added width for publish select to old css

j3nsch committed:
#968 Replaced font-sizes set in px before

j3nsch committed:
#1034 Removed IE CSS for default layout

j3nsch committed:
Merge branch 'v4.8.1' into css996

j3nsch committed:
#1089 Removed doi-import.css

j3nsch committed:
#1089 Special handling of CSS width for some fields

j3nsch committed:
#1089 Special width for some fields in old CSS

j3nsch committed:
#1089 Coding style fix

j3nsch committed:
#1082 Replaced label attribute with title for select-elements

j3nsch committed:
#1082 Fixed unit test after change

vgerlach committed:
Update base.css

Farbliche Hervorhebung der Kennzeichnung einer Testumgebung

vgerlach committed:
GND- und ORCID-Logos fuer Personen

j3nsch committed:
Merge pull request #1103 from vgerlach/patch-14

Update base.css

j3nsch committed:
#996 Margins for red APPLICATION_ENV header

vgerlach committed:
Merge branch 'OPUS4:css996' into css996

j3nsch committed:
Merge pull request #1105 from vgerlach/css996

Logos fuer GND und ORCID

j3nsch committed:
#1106 Center logos with text

vgerlach committed:
Update frontdoor.css

New Twitter/X Logo

vgerlach committed:
new Fontawesome files for folder css

Fontawesome Version 6.4.2

vgerlach committed:
new Files for Fontawesome in folder webfonts

Fontawesome version 6.4.2

vgerlach committed:
Update fontawesome.min.css

Fontawesome version 6.4.2

vgerlach committed:
Update regular.css

Fontawesome version 6.4.2

vgerlach committed:
Update solid.css

Fontawesome version 6.4.2

vgerlach committed:
Update solid.min.css

Fontawesome version 6.4.2

vgerlach committed:
Update to Fontawesome 6.4.2

Modified files to current version

j3nsch committed:
Merge pull request #1107 from vgerlach/patch-15

Update Fontawesome to version 6.4.2 and modification of Twitter/X logo

vgerlach committed:
Update base.css with icon for account

if logged in, this icon will be shown for login-bar element account

vgerlach committed:
Update opus.css - add word break in titles

add word break if the title (h2) is too long

j3nsch committed:
Merge pull request #1116 from vgerlach/patch-16

Update base.css with icon for account

j3nsch committed:
Merge pull request #1117 from vgerlach/patch-17

Update opus.css - add word break in titles

alw-bsz committed:
Merge pull request #7 from OPUS4/v4.8.1

Snychronizing with OPUS4/application/v4.8.1

alw-bsz committed:
#1049: Adding publish templates, translations, and ini-params for new doctypes

alw-bsz committed:
#1142: Harmonizing existing doctypes

alw-bsz committed:
Merge branch 'v4.8.1' of https://github.com/alw-bsz/application into publTemp1049

alw-bsz committed:
Merge pull request #9 from OPUS4/v4.8.1

Synchronizing with application/v4.8.1

alw-bsz committed:
Merge branch 'publTemp1049' of https://github.com/alw-bsz/application into publTemp1049

alw-bsz committed:
#1049: Correcting DocumentTypesTest

j3nsch commented on application/configs/doctypes/article.xml:
```diff
@@ -24,57 +24,62 @@
- * @author      Doreen Thiede <thiede@zib.de>
- * @author      Jens Schwidder <schwidder@zib.de>
- * @copyright   Copyright (c) 2008-2017, OPUS 4 development team
+ * @copyright   Copyright (c) 2023, OPUS 4 development team
```
Das Copyright Jahr wird nicht mehr aktualisiert. Die Datei wurde ursprünglich in 2008 erstellt. Diese Jahreszahl bleibt. Ich werde das in allen betroffenen Dateien selber korrigieren.

j3nsch commented:
Die gebrochenen Tests können angepasst werden.

    Frontdoor_Model_HtmlMetaTagsTest::testGetMappingConfig

Die Zahl der erwarteten Mappings kann angepasst werden.

    Publish_FormControllerTest::testCheckActionWithValidPostAndSendButton
    Publish_FormControllerTest::testCheckActionWithValidPostAndSendButtonAndAllRequiredFields
    Publish_Model_ExtendedValidationTest::testPersonsEmailNotificationWithInvalidData

Die Probleme haben vermutlich alle mit entfernten E-Mail Feldern zu tun. Insbesondere für den ValidationTest ist es vielleicht sinnvoll auf den Dokumenttyp **all** zu wechseln, um diese Funktionalität weiterhin zu testen, für den Fall, dass lokal die E-Mail Felder weiterhin gewünscht sind.

Wir sollten für Entfernungen aus existierenden Dokumenttypen separate Tickets anlegen. Ein Ticket pro entfernten Feld sollte reichen, für alle betroffenen Dokumenttypen. Die Tickets dienen in erster Linie der Dokumentation. Es können auch mehrere zusammengehörende Felder in einem Ticket bearbeitet werden.

Ich kann die Tests auch fixen, aber dann würde ich den Code in den Branch **doctypes1049** übernehmen. Im Augenblick kann ich aber noch auf weitere Anpassungen warten.

extracts committed:
#1144 Adds a configuration option to exclude enrichment fields from indexing

alw-bsz commented:
(Wie) Kann man einzelne Tests testen? Beispielsweise um zu sehen, ob die Umstellung auf den Dokumenttyp funktioniert.

j3nsch commented:
Lokal? Du kannst einen Pfad angeben, auch für eine konkrete Testdatei.

    composer test tests/modules/oai

Einzelne Tests innerhalb einer Testdatei sind komplizierter. Nicht vergessen die Datenbank zurück zu setzen, da manche Tests auch durch Veränderungen in den Testdateien fehlschlagen. Nicht alle Tests räumen sauber auf, insbesondere, wenn es Fehler gab.

    ant reset-testdata

alw-bsz committed:
Merge pull request #10 from OPUS4/v4.8.1

Synchronizing with OPUS4/application/v4.8.1

alw-bsz committed:
#1049, #1142: fixes tests for new and revisioned doctypes

alw-bsz committed:
#1049, #1142: Some fine tuning of the publish forms

extracts committed:
#1151 When using Vagrant, allow Solr calls from any network interface

This allows the Solr 9.x admin UI to be reached from the host OS via http://localhost:9983

alw-bsz committed:
#1049, #1142: Correcting/adding IdentifierIsbn

alw-bsz committed:
#1131: Adds ISMN to publish form and frontdoor

j3nsch committed:
#1154 Allow any identifier type in Publish form

j3nsch committed:
Merge pull request #1155 from OPUS4/bug1154

#1154 Allow any identifier type in Publish form

j3nsch committed:
Merge pull request #1152 from OPUS4/SolrAdminUI1151

Re-enables the Solr admin UI with Solr 9.x

j3nsch committed:
Merge pull request #1145 from OPUS4/excludeFieldSolr1144

Exclude blacklisted enrichment fields from Solr index

alw-bsz committed:
Merge pull request #11 from OPUS4/v4.8.1

Synchronizing with OPUS4/application/v4.8.1

j3nsch committed:
Merge branch 'v4.8.1' into css996

j3nsch committed:
#996 Fixed coding style

j3nsch committed:
#1156 Added basics for UI setup page (empty)

j3nsch committed:
#996 Option to enable old CSS for all pages

j3nsch committed:
#996 Disable legacy CSS as default

j3nsch committed:
#1156 Restrict access to appearance setup

j3nsch committed:
#996 Use variable for theme-vars CSS file

j3nsch committed:
#1156 Fix coding style

j3nsch committed:
#996 Added option themes.selectedTheme

j3nsch committed:
Depend on development version of opus4-import

j3nsch committed:
Fixed opus4-search dependency

j3nsch committed:
#1156 Render form without PHTML file

j3nsch committed:
#1158 config.xml is modified not replaced anymore

j3nsch committed:
#996 Added TODO for later review

j3nsch committed:
#1156 Fixed coding style

j3nsch committed:
#1156 Save selected theme option

j3nsch committed:
#1156 Fixed coding style issue

j3nsch committed:
Merge pull request #997 from OPUS4/css996

Neues, moderneres CSS für OPUS 4

alw-bsz committed:
Merge pull request #12 from alw-bsz/v4.8.1

Synchronizing with v4.8.1

alw-bsz committed:
Merge branch 'ismn1131' into publTemp1049
